idl/github.com/uber/cadence/history.thrift

```diff
@@ -53,6 +53,7 @@ struct GetWorkflowExecutionNextEventIDResponse {
   10: optional i64 (js.type = "Long") eventId
   20: optional string runId
   30: optional shared.TaskList tasklist
+  40: optional bool isWorkflowExecutionRunning
```

Comment:
GetWorkflowExecutionNextEventID returns 'EntityNotExistsError' error to communicate if the workflow is not running.

Reply:
first there should be no performance cost to piggy back one additional field
this is optimized for cases that a new event is added to the history, and with this event, the workflow is closed.

Reply:
Mutable state for workflow execution is immediately deleted when the workflow completes.  Later calls to GetWorkflowExecutionNextEventID should return EntityNotExists.  My guess is any check for IsWorkflowRunning within this method will always return true, or you will get EntityNotExists on loading the workflow execution when it is completed.

Reply:
take RespondActivityTaskCanceled API on the history engine as example: https://github.com/uber/cadence/blob/master/service/history/historyEngine.go

here, the workflow execution context is loaded:
https://github.com/uber/cadence/blob/master/service/history/historyEngine.go#L1145
here, whether the workflow status (running or not) is checked:
https://github.com/uber/cadence/blob/master/service/history/historyEngine.go#L1171

so workflow execution context can be loaded when the workflow is not running.

the GetWorkflowExecutionNextEventID API on the history engine: https://github.com/uber/cadence/blob/master/service/history/historyEngine.go#L284 
does not check whether the workflow is running or not, just simply return the next event ID.

so, my guess, without returning whether the workflow is running, it may take sometime before the workflow execution context is not loadable (causing the EntityNotExistsError)